azure-linux.dockerfile

```
@@ -8,7 +8,11 @@ FROM $IMAGE
 
 # ca-certificates: Azure Linux base image does not contain Mozilla CA certificates, install it to prevent CERTIFICATE_VERIFY_FAILED errors, see https://github.com/Azure/azure-cli/issues/26026
 # jq: It's widely used for parsing JSON output in Azure CLI and has a small size. See https://github.com/Azure/azure-cli/issues/29830
-RUN --mount=type=bind,target=/azure-cli.rpm,source=./docker-temp/azure-cli.rpm tdnf install ca-certificates jq /azure-cli.rpm -y && tdnf clean all && rm -rf /var/cache/tdnf
+RUN --mount=type=bind,target=/azure-cli.rpm,source=./docker-temp/azure-cli.rpm \
+    tdnf install ca-certificates jq /azure-cli.rpm -y && \
+    # Run az version to generate the commandIndex.json and speed up the following az commands
+    az version && cd ~/.azure && ls | grep -v commandIndex.json | xargs rm -rf && \
```

Consider using `--invert-match` for better readability:

```suggestion
    az version && cd ~/.azure && ls | grep --invert-match commandIndex.json | xargs rm -rf && \
```